src/Compilers/Core/MSBuildTask/AssemblyResolution.cs

```diff
@@ -85,6 +102,34 @@ private static bool TryRedirect(AssemblyName name, byte[] token, int major, int
             return false;
         }
 
+        private static readonly string s_assemblyLocation = Utilities.TryGetAssemblyPath(typeof(AssemblyResolution).GetTypeInfo().Assembly);
+        private static Assembly TryRedirectToRuntimesDir(AssemblyName name)
```

Comment:
Do we have a bug tracking the undo'ing of this at a later date?

Reply:
Will we actually undo it? I'm not sure MSBuild or CoreCLR are going to do anything about this.

Reply:
The solution can't be this code though. We need to file a bug with CoreClr to get them to tell us the right way to do this, or provide an assembly load context that has rationale behavior.

Reply:
Filed https://github.com/dotnet/coreclr/issues/15982. I'll add a link

Reply:
Also filed https://github.com/dotnet/roslyn/issues/24397